gms_app/models.py

```diff
@@ -2,36 +2,84 @@
 
 
 class Garden(models.Model):
-    name = models.CharField(max_length=50)
-    description = models.TextField(blank=True)
+    name = models.CharField(max_length=50, verbose_name="Garden Name", help_text="Enter the name of the garden")
+    description = models.TextField(blank=True, help_text="Enter a brief description of the garden")
 
     def __str__(self):
         return self.name
 
+    class Meta:
+        verbose_name = "Garden"
+        verbose_name_plural = "Gardens"
+        ordering = ["name"]
+
 
 class GardenBed(models.Model):
-    garden = models.ForeignKey(Garden, blank=True, on_delete=models.DO_NOTHING, null=True)
-    name = models.CharField(max_length=100)
-    description = models.TextField(blank=True)
-    location = models.JSONField(default=dict)
+    garden = models.ForeignKey(Garden, blank=True, null=True, on_delete=models.SET_NULL, related_name="beds", help_text="Select the garden this bed belongs to")
+    name = models.CharField(max_length=100, verbose_name="Bed Name", help_text="Enter the name of the garden bed")
+    description = models.TextField(blank=True, help_text="Enter a brief description of the garden bed")
 
     def __str__(self):
         return self.name
 
+    class Meta:
+        verbose_name = "Garden Bed"
+        verbose_name_plural = "Garden Beds"
+        ordering = ["name"]
+
+    def add_plant(self, plant, location):
+        return Planting.objects.create(
+            plant=plant,
+            garden_bed=self,
+            location=location,
+        )
+
 
 class Plant(models.Model):
-    garden_bed = models.ManyToManyField(GardenBed, blank=True)
-    name = models.CharField(max_length=50, unique=True)
-    description = models.TextField(blank=True)
-    location = models.JSONField(default=dict)
+    """Represents the Plant "type" that can exist outside of a plant bed
+    """
+    # The name being unique causes a unique constraint error if we do a direct
+    # many to many relationship with the garden bed
+    name = models.CharField(max_length=50, unique=True, verbose_name="Plant Name", help_text="Enter the name of the plant")
```

Comment:
The Plant name being unique was the big culprit for the uniqueness issues the test had

Reply:
Ah! So this was the root of my problems. Unique constraint was persisted from Plant to linking table. Got it!

Reply:
I need to correct my assumption. After sleeping on it, I realized I interpreted the error wrong, but we still want to have the `Planting` join model that has the location. 

You could rewrite the test to look like this so the test will pass. You can't create a new `Plant` instance with the same name, that stands as the uniqueness constraint issue:
```
@pytest.mark.django_db()
def test_allow_same_plant_on_bed():

    garden = models.Garden.objects.filter(name="Test Garden").first()
    beds = models.GardenBed.objects.filter(garden=garden)
   # get the plant, not create a new one with duplicate name
    plant = models.Plant.objects.get(name="Test Plant 1")
    plant.garden_bed.add(*beds)
```

However, if you inspect `plant` then you will see there is still only one relation to "Test Plant 1" to beds[0]. No matter how many times you call `plant.garden_bed.add(bed) for any given bed, it won't create multiple relations between a plant instance and a garden bed instance.